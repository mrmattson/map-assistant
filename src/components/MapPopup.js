import React from 'react'
import PropTypes from 'prop-types'

function MapPopup ({ uniqueId, text }) {
  return (
    <div className="MapPopup" id={uniqueId}>
      {text}
    </div>
  )
}

MapPopup.propTypes = {
  uniqueId: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired
}

export default MapPopup
