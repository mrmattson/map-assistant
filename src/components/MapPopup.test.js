/* eslint-env jest */

import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import MapPopup from './MapPopup'

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(
    <MapPopup
      uniqueId="abc"
      text="def"
    />, div)
})

it('clean snapshot', () => {
  const component = renderer.create(
    <MapPopup
      uniqueId="abc"
      text="def"
    />
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
