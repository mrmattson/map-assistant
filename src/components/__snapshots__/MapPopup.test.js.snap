exports[`test clean snapshot 1`] = `
<div
  className="MapPopup"
  id="abc">
  def
</div>
`;
